### ISSUE TITLE
How to make this work with File scalar in graphql-yoga?

### ISSUE THREAD

**Issue by taybin** [eddeee888/graphql-code-generator-plugins#230]
My project requires working with multi-part uploads and I would like to use graphql-yoga's `scalar File` feature.  I don't see how I can suppress the creation of a File resolver with this plugin though `scalarOverrides` and instead use the built-in File type that graphql-yoga is passing in.

Am I missing something or is this not possible at this point in time?

**Comment by taybin** [comment]
I think the fix is to add `File` to `isNativeNamedType()`.

**Comment by eddeee888** [comment]
Hi @taybin , could you please try out this alpha version?

```shell
yarn add -D @eddeee888/typescript-resolver-files@pr240-run411-1
```

And the new `definteConfig` for your use case may look like this:

```ts
defineConfig({
  resolverGeneration: {
    query: '*',
    mutation: '*',
    subscription: '*',
    scalar: '!*.File', // Generate: all Scalar resolvers in all modules, except for the scalar named `File`
    object: '*',
    union: '',
    interface: '',
  },
})
```

Here's the WIP PR: https://github.com/eddeee888/graphql-code-generator-plugins/pull/240
I'll follow up with more documentation, examples and use cases soon

**Comment by taybin** [comment]
Will do, thanks!

**Comment by taybin** [comment]
@eddeee888 Can you explain what the object, union, and interface settings do?  I don't get the link between those and resolvers.

**Comment by eddeee888** [comment]
Sure! Maybe the name `resolverGeneration` is misleading (something we can change before v1). This is the option that tells preset what _files_ to generate (maybe `fileGeneration` is a better name).

---

Given this schema:
```graphql
# src/schema/base/schema.graphql
interface Error  {
  # ...
}


# src/schema/user/schema.graphql
type User {
  # ...
}

# src/schema/book/schema.graphql
type BookError implements Error {
  # ...
}

union BookResult = BookOk | BookError
```

Then, `object` option will be responsible for:
- `src/schema/user/resolvers/User.ts`. Normalized name (*) of this file is `user.User`
- `src/schema/book/resolvers/BookError.ts`. Normalized name is `book.BookError`
- `src/schema/book/resolvers/BookOk.ts`. Normalized name is `book.BookOk`

`interface` option be be resonspible for:
- `src/schema/base/resolvers/Error.ts`. Normalized name is `base.Error`

`union`  option will be responsible for:
- `src/schema/book/resolvers/BookResult.ts`. Normalized name is `book.BookResult`

`'*'` will generate every file of the relevant type. `''` won't generate any files.

---

(*) Normalized name is important (and will be documented ) as we use can use wildcard to choose what we want to generate. 

Example: some graphs have a standard to wrap results in wrappers e.g. object types ending with `Ok` or `Error`. These usually don't have logic in them but are generated anyways by the preset. `resolverGeneration` allows us to avoid generating these wrapper types:

```ts
resolverGeneration: {
  // ... other config
  object: ['!*Result', '!*Error']
}
```

---

Right now, the default `resolverGeneration` option is `recommended` which generates all files except for unions and interfaces i.e.

```ts
resolverGeneration: {
  query: '*',
  mutation: '*',
  subscription: '*',
  scalar: '*',
  object: '*',
  union: '',
  interface: '',
},
```

**Comment by taybin** [comment]
This suppresses the File resolver generation.  I haven't tested it yet, but:

```typescript
      scalarsOverrides: {
        File: {
          type: 'File',
        },
      },
```

seems to be how to get the appropriate type generated for graphql-yoga.

**Comment by taybin** [comment]
Oh, last bit of feedback, typescript didn't know the new type for `resolverGeneration`.  Not sure if you've set that up yet or not.

**Comment by eddeee888** [comment]
Thanks @taybin !

Once this is released, I'll look into adding `File`-related config into GraphQL Yoga guides.

For types, I can see options being recommended if an object is used:

<img width="1283" alt="Screenshot 2024-03-23 at 8 45 12 am" src="https://github.com/eddeee888/graphql-code-generator-plugins/assets/33769523/f8c7d5ee-07f0-4164-baf4-02a93eff37d0">

I cannot find an easy way to type the expected values of each option though as it's a globby pattern for our GraphQL types. If you know how, let me know!

**Comment by eddeee888** [comment]
Hi @taybin ,

This is now available in [v0.8](https://github.com/eddeee888/graphql-code-generator-plugins/releases/tag/%40eddeee888%2Fgcg-typescript-resolver-files%400.8.0). Thank you for collaborating with me!
